preproc/src/preproc.rs

```diff
@@ -35,24 +40,75 @@ impl Preprocessor for Pandocs {
     }
 
     fn run(&self, _: &PreprocessorContext, mut book: Book) -> Result<Book, Error> {
```

Comment:
Also, shouldn't we add some simple unit testing to this logic?

Reply:
I'm not sure what you'd want to unit-test here, and not super hot on trying to build a mdBook book from scratch for testing purposes—to what extent is that stable? I'd like to avoid churn there.